lib/inspec/base_cli.rb

```diff
@@ -71,24 +73,58 @@ def self.exec_options
         desc: 'Write out a lockfile based on this execution (unless one already exists)'
       option :backend_cache, type: :boolean,
         desc: 'Allow caching for backend command output.'
+      option :show_progress, type: :boolean,
+        desc: 'Show progress while executing tests.'
     end
 
     def self.default_options
       {
         exec: {
+          'reporter' => {
+            'cli' => nil,
+          },
+          'show_progress' => false,
           'color' => true,
           'create_lockfile' => true,
           'backend_cache' => false,
         },
       }
     end
 
+    def self.clean_reporters(opts)
+      # Parse report options from CLI
+      if opts['reporter'].is_a?(Array)
+        stdout = 0
+        reports = {}
+        opts['reporter'].each do |report|
+          k, v = report.split(':')
+          reports[k] = v
+          stdout += 1 if v.nil? || v == '-'
+        end
+
+        raise ArgumentError, 'The option --reporter can only have a single report outputting to stdout.' if stdout > 1
+
+        opts['reporter'] = reports
+      end
+
+      # Catch anyone using the legacy --format option
+      unless opts['format'].nil?
+        opts['reporter'] = { opts['format'] => nil }
+        opts.delete('format')
+      end
+
+      # Catch any dynamic inspec runners or tests
+      opts['reporter'] = { 'cli' => nil } if opts['reporter'].nil?
+
+      opts
+    end
+
     private
 
     # helper method to run tests
     def run_tests(targets, opts)
       o = opts.dup
-      log_device = opts['format'] == 'json' ? nil : STDOUT
+      log_device = o[:reporter]&.keys&.grep(/json/)&.any? ? nil : STDOUT
```

Comment:
This will match `json`, `rspec-json`, and `json-min`, right? Is that intended?  I think that's probably OK. We probably want the same treatment for `junit` too?

A future improvement on this might be to have a method on each reporter that we can call called `suppress_log_output?` or something similar so we do this more programmatically than just string matching.